### PR TITLE
[Markdown] New frontmatter injection feature!

### PR DESCRIPTION
**Pull request by bholmesdev**
#### What kind of changes does this PR include?

- New or updated content

#### Description

- Add new "injecting frontmatter" section to markdown configuration
- Include new reading time tutorial 🚀
- Remove reading time tutorial from `rawContent` docs
- Remove summary tutorial from `compiledContent` docs
- Move `compiledContent` warnings to separate legacy callout

### CODE REVIEW

File: src/pages/en/guides/markdown-content.md [hunk 1]
```diff
@@ -419,6 +395,97 @@ By default, Astro comes with [GitHub-flavored Markdown](https://github.com/remar
     };
     ```
 
+### Injecting frontmatter
+
+You may want to add frontmatter properties to your markdown files programmatically. By using a [remark or rehype plugin](#markdown-plugins), you can generate these properties based on a file's contents.
```
Comment (sarah11918) [inline src/pages/en/guides/markdown-content.md:400]:
```suggestion
You may want to add frontmatter properties to your Markdown files programmatically. By using a [remark or rehype plugin](#markdown-plugins), you can generate these properties based on a file's contents.
```

File: src/pages/en/guides/markdown-content.md [hunk 2]
```diff
@@ -419,6 +395,97 @@ By default, Astro comes with [GitHub-flavored Markdown](https://github.com/remar
...
+
+export default {
+  markdown: {
+    remarkPlugins: [exampleRemarkPlugin],
+  },
+};
+
+```
+
+...every markdown file will have `customProperty` in its frontmatter! This is available when [importing your markdown](#importing-markdown) and from [the `Astro.props.frontmatter` property in your layouts](#markdown-layouts).
```
Comment (sarah11918) [inline src/pages/en/guides/markdown-content.md:428]:
```suggestion
...every Markdown file will have `customProperty` in its frontmatter! This is available when [importing your markdown](#importing-markdown) and from [the `Astro.props.frontmatter` property in your layouts](#markdown-layouts).
```

File: src/pages/en/guides/markdown-content.md [hunk 3]
```diff
@@ -419,6 +395,97 @@ By default, Astro comes with [GitHub-flavored Markdown](https://github.com/remar
...
+
+#### Example: calculate reading time
+
+You can use a [remark plugin](https://github.com/remarkjs/remark) to add a reading time to your frontmatter. We recommend a couple helper packages:
```
Comment (sarah11918) [inline src/pages/en/guides/markdown-content.md:432]:
```suggestion
You can use a [remark plugin](https://github.com/remarkjs/remark) to add a reading time to your frontmatter. We recommend two helper packages:
```
(Simplified language is generally easier for a wider audience while following instructions.)

File: src/pages/en/guides/markdown-content.md [hunk 4]
```diff
@@ -419,6 +395,97 @@ By default, Astro comes with [GitHub-flavored Markdown](https://github.com/remar
...
+    const textOnPage = toString(tree);
+		const readingTime = getReadingTime(textOnPage);
+    // readingTime.text will give us minutes read as a friendly string,
+    // i.e. "3 min read"
+		data.astro.frontmatter.minutesRead = readingTime.text;
+	};
+}
+```
+
+Once you apply this plugin your config:
```
Comment (sarah11918) [inline src/pages/en/guides/markdown-content.md:458]:
```suggestion
Once you apply this plugin to your config:
```
Personal nit: Feel free to ignore, but I might tend to say "add this plugin to your config" as I think we do often talk about "adding to config" vs "applying to config" throughout the docs. Also, we've "applied" in the step above so "add" mixes it up! Variety is fun!  🥳

File: src/pages/en/guides/markdown-content.md [hunk 5]
```diff
@@ -419,6 +395,97 @@ By default, Astro comes with [GitHub-flavored Markdown](https://github.com/remar
...
+
+export default {
+  markdown: {
+    remarkPlugins: [remarkReadingTime],
+  },
+};
+
+```
+
+...all markdown documents will have a calculated `minutesRead`. You can use this to apply an "X min read" banner from a [markdown layout](#markdown-layouts), for instance:
```
Comment (sarah11918) [inline src/pages/en/guides/markdown-content.md:472]:
```suggestion
...all Markdown documents will have a calculated `minutesRead`. You can use this to include an "X min read" banner in a [markdown layout](#markdown-layouts), for instance:
```
Again, personal preference: I think this turn of phrase might feel a little more consistent with having a code sample showing this banner *in* the layout component below. It feels more natural to me. But, the only true correction is the capitalization of Markdown!

**Review by sarah11918 — COMMENTED** [review]
The only thing I don't love is how long this page is getting... 😅  BUT, that is a separate issue!

Just capitalized some Markdowns, and "a couple things" I'd suggest very minor changes to, but wont hold you to.

Thanks for this amazing example, @bholmesdev, that I know is gonna get a lotta use, and a lotta love!  🚀

**Review by Pukimaa — APPROVED** [review]
LGTM! Love this new feature!

**Comment by Pukimaa** [comment]
Oopsie! For some reason I get this error when using your example:
```
file:///home/jaron/Dokumente/coding/web-astro/astro.config.mjs:4
import { remarkReadingTime } from './remark-reading-time.js';
         ^^^^^^^^^^^^^^^^^
SyntaxError: Named export 'remarkReadingTime' not found. The requested module './remark-reading-time.js' is a CommonJS module, which may not support all module.exports as named exports.
CommonJS modules can always be imported via the default export, for example using:

import pkg from './remark-reading-time.js';
const { remarkReadingTime } = pkg;

    at ModuleJob._instantiate (node:internal/modules/esm/module_job:127:21)
    at async ModuleJob.run (node:internal/modules/esm/module_job:193:5)
    at async Promise.all (index 0)
    at async ESMLoader.import (node:internal/modules/esm/loader:337:24)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by Pukimaa** [comment]
Got a fix:
1. Export the Plugin Function as a default
2. Use the `.mjs` extension
3. Don't import the plugin as a named export. Use something like:
`import remarkReadingTime from './remark-reading-time.mjs';`

**Comment by sarah11918** [comment]
Heads up, @bholmesdev, I should also note that Sarah still needs to sit down and familiarize herself with how we now author code snippets, thanks to @hippotastic!  🥳 

So, let's be sure to wait to merge until we've fancied-up the actual code snippets themselves.

**Comment by bholmesdev** [comment]
@Pukimaa Good catch! Adding `.mjs` to the file extension + import statement. Doesn't look like the named export is an issue though; using that in our test suites right now

**Comment by bholmesdev** [comment]
cc @hippotastic fancied up the line highlighting myself too 😁 Love this highlighting API by the way, it's very easy to use!

**Comment by hippotastic** [comment]
> cc @hippotastic fancied up the line highlighting myself too 😁 Love this highlighting API by the way, it's very easy to use!

Glad to hear that! By the way, in multiple of your code snippets, you're showing code to insert into an existing file. You can prefix your highlighting terms with an `ins=` prefix to change the marker type to a green insertion marker. For example, this:

````md
```astro title="src/layouts/BlogLayout.astro" "const { minutesRead } = Astro.props.content;" "<p>{minutesRead}</p>"
````

Could become that:

````md
```astro title="src/layouts/BlogLayout.astro" ins="const { minutesRead } = Astro.props.content;" ins="<p>{minutesRead}</p>"
````

And the highlights would turn green.

Also, if you're intending to highlight entire lines anyway, you can target them by line number instead of having to copy the entire line contents into a search term. For example, this syntax marks lines 2 and 7 as inserted:

````md
```astro title="src/layouts/BlogLayout.astro" ins={2,7}
---
const { minutesRead } = Astro.props.content;
---
<html>
  <head>...</head>
  <body>
    <p>{minutesRead}</p>
    <slot />
  </body>
</html>
```
````

**Pull request merged**